sdk/eventhub/Azure.Messaging.EventHubs/tests/Authorization/SharedAccessSignatureTests.cs

```diff
@@ -423,53 +423,53 @@ public void ExtendExpirationUpdatesTheExpirationTime()
             var composedSignature = new SharedAccessSignature("amqps://some.namespace.com/hubName", keyName, key, validFor);
             var parsedSignature = new SharedAccessSignature(composedSignature.ToString(), keyName) as SharedAccessSignature;
 
-            parsedSignature.ExtendExpiration(extendBy);
+            parsedSignature = parsedSignature.CloneAndExtendExpiration(extendBy);
             Assert.That(parsedSignature.SignatureExpiration, Is.EqualTo(expiration).Within(TimeSpan.FromSeconds(5)));
         }
 
         /// <summary>
-        ///   Verifies functionality of the <see cref="ParseSignature" />
+        ///   Verifies functionality of the <see cref="SharedAccessSignature.Clone" />
         ///   method.
         /// </summary>
         ///
         [Test]
-        public void ParseProducesCorrectValues()
+        public void CloneProducesACopy()
         {
             var resource = "amqps://some.namespace.com/hubName";
             var keyName = "rootShared";
             var key = "ABC123FFF333";
             var validFor = TimeSpan.FromMinutes(30);
-            DateTimeOffset expiration = DateTimeOffset.UtcNow.Add(validFor);
             var signature = new SharedAccessSignature(resource, keyName, key, validFor);
-            (string KeyName, string Resource, DateTimeOffset ExpirationTime) parsed = ParseSignature(signature.ToString());
+            SharedAccessSignature clone = signature.CloneAndExtendExpiration(TimeSpan.FromSeconds(5));
 
-            Assert.That(parsed, Is.Not.Null, "There should have been a result returned.");
-            Assert.That(parsed.Resource, Is.EqualTo("amqps://some.namespace.com/hubName"), "The resource should match.");
-            Assert.That(parsed.KeyName, Is.EqualTo(keyName), "The key name should have been parsed.");
-            Assert.That(parsed.ExpirationTime, Is.EqualTo(expiration).Within(TimeSpan.FromSeconds(5)), "The expiration should be parsed.");
+            Assert.That(clone, Is.Not.Null, "There should have been a copy produced.");
+            Assert.That(clone, Is.Not.SameAs(signature), "The clone should be a unique instance.");
+            Assert.That(clone.Resource, Is.EqualTo(signature.Resource), "The resource should match.");
+            Assert.That(clone.SharedAccessKeyName, Is.EqualTo(signature.SharedAccessKeyName), "The key name should match.");
+            Assert.That(clone.SharedAccessKey, Is.EqualTo(signature.SharedAccessKey), "The key should match.");
+            Assert.That(clone.SignatureExpiration, Is.Not.EqualTo(signature.SignatureExpiration), "The expiration should not match; it should have been extended.");
```

Comment:
It may be good to verify that the expiration really got updated with the expected value. Something like we do in `ParseProduceCorrectValues`.

Reply:
Fine.  I'll stop being lazy.   😄